auth/client/src/main/java/org/wildfly/security/auth/client/AuthenticationContext.java

```diff
@@ -244,8 +244,15 @@ public AuthenticationContext withoutSsl(int idx) throws IndexOutOfBoundsExceptio
     RuleNode<SecurityFactory<SSLContext>> sslRuleMatching(URI uri, String abstractType, String abstractTypeAuthority) {
         RuleNode<SecurityFactory<SSLContext>> node = this.sslRules;
         while (node != null) {
-            if (node.getRule().matches(uri, abstractType, abstractTypeAuthority)) return node;
-            node = node.getNext();
+            if (uri == null) {
+                if (node.getRule().equals(MatchRule.ALL)) {
```

Comment:
I assume this is needed here as the default SSLContext implementation doesn't know the destination it is being used for so we have no URI - are there any compatibility issues by matching ALL possibly sooner than we used to or do we feel this is safe?

Reply:
Matching ALL sooner than we used to would happen only if null URI was passed here. Which was not possible before my changes because this method is only called in [getSSLContextFactory](https://github.com/wildfly-security/wildfly-elytron/blob/1.x/auth/client/src/main/java/org/wildfly/security/auth/client/AuthenticationContextConfigurationClient.java#L235) and there was a [null check for URI](https://github.com/wildfly-security/wildfly-elytron/blob/1.x/auth/client/src/main/java/org/wildfly/security/auth/client/AuthenticationContextConfigurationClient.java#L233). I removed this null check in this PR. 

I think backwards incompatibility can happen only if users were relying on the IllegalArgumentException to happen by passing null to the [getSSLContextFactory](https://github.com/wildfly-security/wildfly-elytron/blob/1.x/auth/client/src/main/java/org/wildfly/security/auth/client/AuthenticationContextConfigurationClient.java#L232) method. Is this something we should be worried about?

Btw to avoid this I can introduce new config parameter for default SSL context meant only for this provider. But that would not leverage existing possibility to configure default SSLContext.